src/aap_eda/core/models/user.py

```diff
@@ -32,6 +32,10 @@ class User(AbstractUser):
     roles = models.ManyToManyField("Role", related_name="users")
     modified_at = models.DateTimeField(auto_now=True, null=False)
 
+    def get_awx_tokens(self):
```

Comment:
@Alex-Izquierdo Can this code be shared and reused with get_awx_token by passing in the user_id?

Reply:
@mkanoor I'm not following you. What do you want exactly? 
This method extends the user model as a shortcut to fetch the tokens (must be multiple because we don't have the constraint of only one token) and is called in a couple of places for validations. 

get_awx_token was implemented to return only the first one from the user_id and is tied to the consumer. 

Do you want to return the first token or an exception directly from the model?

Reply:
@Alex-Izquierdo I am thinking that code should return an array of tokens, you can do the count check in one place and in the other place take the first token. This same code needs to be called before we start an [activation](https://github.com/ansible/eda-server/blob/0b746b0c2d3d96f28cc7fba1429aace089e30c8b/src/aap_eda/tasks/ruleset.py#L40C1-L40C1), because there is no point in calling activation and then have it come over the web socket to be rejected, we can stop it right away before an activation starts either via the scheduler doing a restart or via the UI. If the number of tokens is > 1 or the number of tokens is 0. We could just stop the activation from starting with an error saying more than 1 token has been found.

get_awx_tokens(user_id)

Reply:
Ok, np. I think I understand what you say about the re usability. 

About the places where we are checking the token, before the PR, we only checked it at the creation of the activation. Now, I have added the check before start the activation but [after initiate the instance](https://github.com/ansible/eda-server/pull/377/files#diff-475356a7e610a5db9004d54074241bde14df30fbfabebc4cdac68aaacba9e8ccR130). I can not move the check where you say because we need an activation_instance created to notice to the user what was the error. 
The check in the websocket is also necessary because there is a time window where the token can be deleted between when the activation_instance start and the websocket connection is established.